File: ts/wam/7900.ts
export const WamAudienceManagement = 7900 as const

export interface AudienceManagementEvent {
  /** field 1, wire `audience_event_surface` */
  audienceEventSurface?: AUDIENCEEVENTSURFACETYPE
  /** field 2, wire `audience_extra_data` */
  audienceExtraData?: string
  /** field 3, wire `audience_management_action` */
  audienceManagementAction?: AUDIENCEMANAGEMENTACTIONTYPE
  /** field 4, wire `audience_predicate_type` */
  audiencePredicateType?: AUDIENCEPREDICATETYPEENUM
  /** field 5, wire `audience_resolution_trigger` */
  audienceResolutionTrigger?: AUDIENCERESOLUTIONTRIGGERTYPE
}

File: rust/wam/7900.rs
pub const WAM_AUDIENCE_MANAGEMENT: u32 = 7900;

#[derive(Debug, Default)]
pub struct AudienceManagementEvent {
    /// field 1, wire `audience_event_surface`
    pub audience_event_surface: Option<AUDIENCEEVENTSURFACETYPE>,
    /// field 2, wire `audience_extra_data`
    pub audience_extra_data: Option<String>,
    /// field 3, wire `audience_management_action`
    pub audience_management_action: Option<AUDIENCEMANAGEMENTACTIONTYPE>,
    /// field 4, wire `audience_predicate_type`
    pub audience_predicate_type: Option<AUDIENCEPREDICATETYPEENUM>,
    /// field 5, wire `audience_resolution_trigger`
    pub audience_resolution_trigger: Option<AUDIENCERESOLUTIONTRIGGERTYPE>,
}